crates/matrix-sdk-ui/src/timeline/event_item/local.rs

```diff
@@ -84,3 +90,21 @@ pub enum EventSendState {
         event_id: OwnedEventId,
     },
 }
+
+/// This type represents the "send progress" of a local event timeline item.
+#[derive(Clone, Debug)]
+pub enum EventSendProgress {
+    /// A media (consisting of a file and possibly a thumbnail) is being
+    /// uploaded.
+    MediaUpload {
```

Comment:
ditto, since this is an enum with a single variant, and that's not ser/de, let's use a struct instead